keadm/cmd/keadm/app/cmd/edge/batchjoin.go

```diff
@@ -0,0 +1,179 @@
+)
+
+type Node struct {
+	IP              string `yaml:"ip"`
+	EdgeNodeName    string `yaml:"EdgeNodeName"`
+	Architecture    string `yaml:"architecture"`
+	Username        string `yaml:"username"`
+	Password        string `yaml:"password"`
+	KubeedgeVersion string `yaml:"version"`
+}
+
+type KeadmParams struct {
+	CloudCoreIPPort string `yaml:"cloudcore-ipport"`
+	Token           string `yaml:"token"`
+	KubeedgeVersion string `yaml:"kubeedge-version"`
+}
+
+type Config struct {
+	KeadmParams KeadmParams `yaml:"keadm_params"`
+	Nodes       []Node      `yaml:"nodes"`
+}
+
+var configFile string
+
+// NewDeprecatedEdgeJoin returns KubeEdge edge join command.
+func NewEdgeBatchjoin() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "batchjoin",
+		Short: "Batch join nodes using a config file",
+		Long:  `This command allows multiple nodes to join a cluster using a specified config file.`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if configFile == "" {
+				fmt.Println("Please provide a config file using -c")
+				os.Exit(1)
+			}
+			fmt.Printf("Joining nodes using config file: %s\n", configFile)
+			processBatchjoin(configFile)
+		},
+	}
+	cmd.Flags().StringVarP(&configFile, "config", "c", "", "Path to config file")
+	return cmd
+}
+func processBatchjoin(cfgFile string) {
+	configData, err := ioutil.ReadFile(configFile)
+	if err != nil {
+		fmt.Printf("Error reading config file: %v", err)
+		return
+	}
+	var cfg Config
+	err = yaml.Unmarshal(configData, &cfg)
+	if err != nil {
+		fmt.Printf("Error unmarshaling config data: %v", err)
+		return
+	}
+	// Batch join edge nodes
+	if err := batchJoinNodes(&cfg); err != nil {
+		fmt.Printf("Failed to batch join nodes: %v", err)
+	}
+}
+func batchJoinNodes(config *Config) error {
+	ctx := context.Background()
+	var wg sync.WaitGroup                        // Used to wait for all goroutines to complete
+	errCh := make(chan error, len(config.Nodes)) // Used to collect errors
+
+	for _, node := range config.Nodes {
+		wg.Add(1) // Each node corresponds to a goroutine
+		go func(n Node) {
+			defer wg.Done() // Ensure Done is called when the goroutine finishes
+			fmt.Printf("Processing node %s \n", n.IP)
+			// Perform download and copy of keadm
+			if err := downloadAndCopyKeadmToNode(ctx, &n); err != nil {
+				errCh <- err // If an error occurs, send it to the error channel
+				fmt.Printf("Processing node %s  downloadAndCopyKeadmToNode error \n", n.IP)
+				return
+			}
+			// Execute the keadm join command
+			if err := executeJoinCommand(ctx, &n, &config.KeadmParams); err != nil {
+				fmt.Printf("Processing node %s  executeJoinCommand error \n", n.IP)
+				errCh <- err // Send to the error channel when an error occurs
+				return
+			}
+			fmt.Printf("Node %s joined successfully\n", n.IP)
+		}(node)
+	}
+
+	// Wait for all goroutines to complete
+	wg.Wait()
+	close(errCh)
+
+	// Check for errors
+	for err := range errCh {
+		if err != nil {
+			return err // 如果有任意一个错误，返回第一个错误
```

Comment:
Different nodes may encounter distinct errors; informing the user about all of them can significantly expedite the debugging process.

Personal suggestion: When an error occurs, in addition to sending the error to the `errCh` channel, also print the error message directly. The error return of `batchJoinNodes` only indicates which nodes encountered errors, without specifying what the errors are, allowing users to debug based on the aforementioned printed error messages.